modules/heat_conduction/src/userobjects/GapFluxModelConduction.C

```diff
@@ -0,0 +1,96 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GapFluxModelConduction.h"
+#include "libmesh/utility.h"
+#include "Function.h"
+
+registerMooseObject("HeatConductionApp", GapFluxModelConduction);
+
+InputParameters
+GapFluxModelConduction::validParams()
+{
+  InputParameters params = GapFluxModelBase::validParams();
+  params.addClassDescription("Gap flux model with a constant conductance");
+  params.addCoupledVar("T", "Temperature");
+  params.addRequiredParam<Real>("gap_conductivity", "Gap conductivity value");
+  params.addParam<FunctionName>(
+      "gap_conductivity_function",
+      "Thermal conductivity of the gap material as a function.  Multiplied "
+      "by gap_conductivity.");
+  params.addCoupledVar("gap_conductivity_function_variable",
+                       "Variable to be used in the gap_conductivity_function in place of time");
+  params.addRangeCheckedParam<Real>(
+      "min_gap", 1e-6, "min_gap>0", "A minimum gap (denominator) size");
+  params.addRangeCheckedParam<unsigned int>(
+      "min_gap_order",
```

Comment:
should the min gap stuff be handled in the constraint maybe? Or is this really only relevant for the conduction flux model?

Reply:
I debated this... It's only used for conduction now, so it can be moved to the constraint if it ever becomes more general. Some of the computations in the constraint are only used for conduction too, but they are related to general problem parameters, so Ieft those in the constraint. The `min_gap` appears to be more conduction model-related.